### PR TITLE
Update proxies.md

### PR DESCRIPTION
**Pull request by RakeshJesadiya**
Example of the declare proxy class from the core code.

## Purpose of this pull request

This pull request (PR) adds the live example of how to add proxy class using di.xml file in Magento 2.

## Affected DevDocs pages

https://devdocs.magento.com/guides/v2.3/extension-dev-guide/proxies.html
https://devdocs.magento.com/guides/v2.4/extension-dev-guide/proxies.html

### CODE REVIEW

File: src/guides/v2.3/extension-dev-guide/proxies.md [hunk 1]
```diff
@@ -73,3 +73,13 @@ With the proxy used in place of `SlowLoading`, the `SlowLoading` class will not
 Because DI configuration is used to inject a proxy, proxies can be dropped in to replace their corresponding classes - or proxy replacements _removed_ - without touching application code.
 
 As a practical example of a proxy, you can see the [StoreManager]({{ site.mage2bloburl }}/{{ page.guide_version }}/app/code/Magento/Store/Model/StoreManager.php){:target="_blank"} class and then see the generated `StoreManager` proxy class.
+
+Real Example from the Magento core code, We are passing `storeManager` argument as proxy to the `Magento\Store\Model\Resolver\Store` class.
+Define StoreManagerInterface as proxy class by adding `Proxy` Class at the end of original class in the `di.xml` file. `Magento\Store\Model\StoreManagerInterface\Proxy` 
+```xml
```
Comment (hguthrie) [inline src/guides/v2.3/extension-dev-guide/proxies.md:79]: Your proposed addition requires some formatting fixes, including inserting a line before the XML snippet. I suggest the following to help with clarification, but will still need verification.

```suggestion
The following excerpt from the Magento code passes the `storeManager` argument as a proxy to the `Magento\Store\Model\Resolver\Store` class. The `StoreManagerInterface` model is defined as a proxy class by adding `Proxy` at the end of original class in the `di.xml` file.

```xml
```

**Review by hguthrie — CHANGES_REQUESTED** [review]

**Comment by RakeshJesadiya** [comment]
@hguthrie I have updated the changes as per your request. Thank you.

**Review by hguthrie — APPROVED** [review]
As far as I can tell, it looks good. I will find someone to double-check. Thanks for the contribution.

File: src/guides/v2.3/extension-dev-guide/proxies.md [hunk 2]
```diff
@@ -73,3 +73,13 @@ With the proxy used in place of `SlowLoading`, the `SlowLoading` class will not
 Because DI configuration is used to inject a proxy, proxies can be dropped in to replace their corresponding classes - or proxy replacements _removed_ - without touching application code.
 
 As a practical example of a proxy, you can see the [StoreManager]({{ site.mage2bloburl }}/{{ page.guide_version }}/app/code/Magento/Store/Model/StoreManager.php){:target="_blank"} class and then see the generated `StoreManager` proxy class.
+
+The following excerpt from the Magento code passes the `storeManager` argument as a proxy to the `Magento\Store\Model\Resolver\Store` class. The `StoreManagerInterface` model is defined as a proxy class by adding `Proxy` at the end of original class in the `di.xml` file.
```
Comment (hguthrie) [inline src/guides/v2.3/extension-dev-guide/proxies.md:77]:
```suggestion
The following excerpt from the Magento code passes the `storeManager` argument as a proxy to the `Magento\Store\Model\Resolver\Store` class. The `StoreManagerInterface` model is defined as a proxy class by the added `Proxy` at the end of the original class in the `di.xml` file.
```

**Review by hguthrie — APPROVED** [review]
After asking someone internal, looks like it needs a little tweak.
@BarnyShergold  Would you like to review this updated example?

**Comment by RakeshJesadiya** [comment]
@hguthrie I have updated the changes. Thank you.

**Comment by hguthrie** [comment]
running tests

**Pull request merged**